File: src/main/java/org/lts/lego_teh_set/evenst/GuildsCounter.java
package org.lts.lego_teh_set.evenst;

import net.dv8tion.jda.api.entities.Activity;
import net.dv8tion.jda.api.events.guild.GuildJoinEvent;
import net.dv8tion.jda.api.events.guild.GuildLeaveEvent;
import net.dv8tion.jda.api.events.session.ReadyEvent;
import net.dv8tion.jda.api.hooks.ListenerAdapter;
import org.lts.lego_teh_set.LegoTehSetApplication;

public class GuildsCounter extends ListenerAdapter {

    private void updateGuildsLengthActivity() {
        LegoTehSetApplication.getShard().setActivity(
                Activity.watching(
                        String.format("%s Guilds", LegoTehSetApplication.getShard()
                                .getGuilds().size())));
    }

    @Override
    public void onReady(ReadyEvent event) {
        this.updateGuildsLengthActivity();
    }

    @Override
    public void onGuildJoin(GuildJoinEvent event) {
        this.updateGuildsLengthActivity();
    }

    @Override
    public void onGuildLeave(GuildLeaveEvent event) {
        this.updateGuildsLengthActivity();
    }

}

File: src/main/java/lego_teh_set_discord_bot/evenst/GuildsCounter.kt
package lego_teh_set_discord_bot.evenst

import net.dv8tion.jda.api.entities.Activity
import net.dv8tion.jda.api.events.guild.GuildJoinEvent
import net.dv8tion.jda.api.events.guild.GuildLeaveEvent
import net.dv8tion.jda.api.events.session.ReadyEvent
import net.dv8tion.jda.api.hooks.ListenerAdapter
import org.example.Main

/**
 * Считает количество гильдий, на которых находится бот.
 */
class GuildsCounter : ListenerAdapter() {

    private fun updateGuildsLengthActivity() {

        Main.getShard().setActivity(Activity.watching("${Main.getShard().guilds.size} Guilds"));
    }

    override fun onReady(event: ReadyEvent) {

        this.updateGuildsLengthActivity()
    }

    override fun onGuildJoin(event: GuildJoinEvent) {

        this.updateGuildsLengthActivity()
    }

    override fun onGuildLeave(event: GuildLeaveEvent) {

        this.updateGuildsLengthActivity()
    }
}